server/graphql/common/update.js

```diff
@@ -0,0 +1,84 @@
+import { get } from 'lodash';
+
+import { mustHaveRole } from '../../lib/auth';
+import { purgeCacheForCollective } from '../../lib/cache';
+import { stripTags } from '../../lib/utils';
+import models from '../../models';
+import { NotFound, ValidationFailed } from '../errors';
+import { idDecode, IDENTIFIER_TYPES } from '../v2/identifiers';
+
+function require(args, path) {
+  if (!get(args, path)) {
+    throw new ValidationFailed(`${path} required`);
+  }
+}
+
+export async function createUpdate(_, args, req) {
+  const CollectiveId = get(args, 'update.account.legacyId');
+  mustHaveRole(req.remoteUser, 'ADMIN', CollectiveId, 'create an update');
+  require(args, 'update.title');
+
+  const collective = await models.Collective.findByPk(CollectiveId);
+
+  if (!collective) {
+    throw new Error('This collective does not exist');
+  }
+
+  const markdown = args.update.markdown ? stripTags(args.update.markdown) : '';
+
+  const update = await models.Update.create({
+    title: args.update.title,
+    markdown,
+    html: stripTags(args.update.html),
+    CollectiveId,
+    isPrivate: args.update.isPrivate,
+    TierId: get(args, 'update.tier.id'),
+    CreatedByUserId: req.remoteUser.id,
+    FromCollectiveId: req.remoteUser.CollectiveId,
+    makePublicOn: args.update.makePublicOn,
+  });
+
+  purgeCacheForCollective(collective.slug);
+  return update;
+}
+
+async function fetchUpdate(id) {
+  const update = await models.Update.findByPk(idDecode(id, IDENTIFIER_TYPES.UPDATE));
```

Comment:
If I understand correctly, since this function is called from both GQLV1 & GQLV2, `id` can be received as either a `string` or an `integer`. We should make sure it works with both.

Reply:
@Betree : I've kept `v1/mutations/updates.js` as it is; https://github.com/opencollective/opencollective-api/blob/comments-on-updates/server/graphql/v1/mutations/updates.js

This one is the one that is triggered for the V1 endpoint. So my understanding is that it will be working for both. I'll remove `v1/mutations/updates.js` in the next PR after this is deployed and we are ready to deprecate the V1 endpoints. 😄

Reply:
But if you call this `fetchUpdate` from V1, won't it be called with the integer ID like that? 

```
fetchUpdate(4242);
```

Reply:
@Betree : Yes you are correct indeed. Sorry my mistake, was confused with this one. Anyways, I made the change to handle both. 😄 Let me know if you see any issues.